jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/CodeWhispererExplorerActionManager.kt

```diff
@@ -198,10 +206,17 @@ interface CodeWhispererActivationChangedListener {
     fun activationChanged(value: Boolean) {}
 }
 
+private fun ToolkitConnection.state(): BearerTokenAuthState =
+    ((this as? AwsBearerTokenConnection)?.getConnectionSettings()?.tokenProvider?.delegate as? BearerTokenProvider)?.state()
+        ?: BearerTokenAuthState.NOT_AUTHENTICATED
+
 fun isCodeWhispererEnabled(project: Project) = with(CodeWhispererExplorerActionManager.getInstance()) {
     checkActiveCodeWhispererConnectionType(project) != CodeWhispererLoginType.Logout
 }
 
 fun isCodeWhispererExpired(project: Project) = with(CodeWhispererExplorerActionManager.getInstance()) {
-    checkActiveCodeWhispererConnectionType(project) == CodeWhispererLoginType.Expired
+    if (checkActiveCodeWhispererConnectionType(project) == CodeWhispererLoginType.Logout) {
+        return@with null
+    }
+    return@with checkActiveCodeWhispererConnectionType(project) == CodeWhispererLoginType.Expired
```

Comment:
return `null` intentionally when the connection state is `un_authenticate`

Reply:
as a followup, I'm thinking it would be better to remove both `isCodeWhispererEnabled` and `isCodeWhispererExpired` and only use `checkActiveCodeWhispererConnectionType`. But will open another PR for that purpose for smaller code change surface